### DISCUSSION TITLE
Regarding maximum and minimum metal routing layer issue.

### DISCUSSION THREAD

**Discussion by its-rakib** [The-OpenROAD-Project/OpenROAD-flow-scripts#857] [General]
For routing I am setting  "MIN_ROUTING_LAYER = M2"   "MAX_ROUTING_LAYER = M4" in the /OpenROAD-flow-scripts/flow/platforms/asap7/config.mk file.
But, In routing stage tool is using M5 and M6 while routing I also tried with  "MIN_ROUTING_LAYER = M2"   "MAX_ROUTING_LAYER = M8" but there is no change in routing, tool is using upto M6. 
Is it a tool issue or I am doing something wrong. It would be great if you help me with the issue.
Thank you for your time.

**Comment by vijayank88** [comment]
Can you check again? Those you're mentioned related to power grid [strategy](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/platforms/asap7/openRoad/pdn/grid_strategy-M2-M5-M7.tcl)
The Min/Max routing layer used for detailed routing. That you can verify in the detail route log. 
Like following message:
```
detailed_route arguments:  -bottom_routing_layer M2 -top_routing_layer M4 -save_guide_updates -verbose 1
```

**Reply by its-rakib** [reply]
Hi, 5_2_TritonRoute.log shows 0 wirelengths for M7 and M8.

![image](https://user-images.githubusercontent.com/127094304/224006989-b5dbbf1b-a127-41e0-8b51-2d7767f58f33.png)

**Reply by vijayank88** [reply]
can you attach full log `5_2_TritonRoute.log`?

**Reply by vijayank88** [reply]
The allocated metal resource till M6 itself tool able to route it, so remaining metal resource un-used for this design. Try with complex design. You may see the log, all metal resource used during that time.

**Reply by its-rakib** [reply]
[5_2_TritonRoute.log](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/10938134/5_2_TritonRoute.log)

**Comment by maliberty** [comment]
I believe the report is for all nets including the power grid.  Do you actually observe signal nets on those layers?

**Reply by its-rakib** [reply]
yes,that was only signal net

**Comment by maliberty** [comment]
What design are you running?

**Reply by its-rakib** [reply]
riscV32i with asap7 pdk

**Comment by maliberty** [comment]
The pins on M5 will be unreachable:
```
export IO_PLACER_H             = M4
export IO_PLACER_V             = M5
```
I changed M5 to M3 and it routes fine.

@eder-matheus we should have a check for pins above the max routing layer (or connect from below with your new drt work)
